src/experimental/theoretic_information.cpp

```diff
@@ -0,0 +1,644 @@
+            float free_prop = meas_outcm[r][0];
+            float occ_prop = meas_outcm[r][1];
+            float un_prop = meas_outcm[r][2];
+
+            if (fr_idx + oc_idx + un_idx == r)
+            {
+                // Searching for the current combination in this level
+                std::vector<map_tuple>::iterator it_comb;
+                it_comb = std::find(tup_vct.begin(), tup_vct.end(), std::make_tuple(fr_idx + 1, oc_idx, un_idx));
+
+                // Free
+                if (it_comb != tup_vct.end())
+                {
+                    acc_prob[it_comb - tup_vct.begin()] += it_out->second * free_prop;
+                }
+                else
+                {
+                    tup_vct.push_back(std::make_tuple(fr_idx + 1, oc_idx, un_idx));
+                    acc_prob.push_back(it_out->second * free_prop);
+                }
+
+                it_comb = std::find(tup_vct.begin(), tup_vct.end(), std::make_tuple(fr_idx, oc_idx + 1, un_idx));
+
+                // Occupied
+                if (it_comb != tup_vct.end())
+                {
+                    acc_prob[it_comb - tup_vct.begin()] += it_out->second * occ_prop;
+                }
+                else
+                {
+                    tup_vct.push_back(std::make_tuple(fr_idx, oc_idx + 1, un_idx));
+                    acc_prob.push_back(it_out->second * occ_prop);
+                }
+
+                it_comb = std::find(tup_vct.begin(), tup_vct.end(), std::make_tuple(fr_idx, oc_idx, un_idx + 1));
+
+                // Unobserved
+                if (it_comb != tup_vct.end())
+                {
+                    acc_prob[it_comb - tup_vct.begin()] += it_out->second * un_prop;
+                }
+                else
+                {
+                    tup_vct.push_back(std::make_tuple(fr_idx, oc_idx, un_idx + 1));
+                    acc_prob.push_back(it_out->second * un_prop);
+                }
+            }
+        }
+        // Inserting the elements into the map
+        for (int k = 0; k < tup_vct.size(); ++k)
+        {
+            map_out.insert(map_pair(tup_vct[k], acc_prob[k]));
+        }
+    }
+    return map_out;
+}
+
+float TeorethicInformation::euclideanDistance(float x_1, float y_1, float x_2, float y_2)
+{
+    /*
+        To calculate the euclidean distance between two points
+    */
+    float diff_x = x_2 - x_1;
+    float diff_y = y_2 - y_1;
+
+    return sqrt(diff_x*diff_x + diff_y*diff_y);
+}
+
+int TeorethicInformation::getSign(int n_1, int n_2)
+{
+    /*
+        To get the sign of an operation, used for Bresenham algorithm
+    */
+    int difference = n_2 - n_1;
+
+    if (difference == 0) { return 0; }
+    else if (difference < 0) { return -1; }
+    else { return 1; }
+}
+
+std::vector<int> TeorethicInformation::getGridPosition(float x, float y)
+{
+    /*
+        To maps the current position into grid coordinates
+    */
+    int x_cell = floor((1 / m_resolution) * x);
```

Comment:
@kmilo7204 nit: consider
```suggestion
    int x_cell = floor(x / m_resolution);
```
Same below.